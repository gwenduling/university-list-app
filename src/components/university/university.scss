.university-component {
  padding: 10px;
  color: var(--color-darker);
  background-color: var(--color-light);

  .name {
    margin: 0;
    font-weight: 700;
  }

  .link {
    color: blue;
    font-size: 12px;
  }
}
