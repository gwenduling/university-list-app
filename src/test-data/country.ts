import { CountryModel } from "../models/countries";
export const mockCountry: CountryModel = {
  alpha2Code: "PH",
  altSpellings: ["PH", "Republic of the Philippines", "Rep√∫blika ng Pilipinas"],
  flag: "https://restcountries.eu/data/phl.svg",
  name: "Philippines",
  nativeName: "Pilipinas",
  count: 0,
  total: 0,
};

export const mockCountry2: CountryModel = {
  alpha2Code: "US",
  altSpellings: ["US", "USA", "United States of America"],
  flag: "https://restcountries.eu/data/phl.svg",
  name: "United States of America",
  nativeName: "United States",
  count: 0,
  total: 0,
};
