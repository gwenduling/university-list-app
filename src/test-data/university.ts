import { UniversityModel } from "../models/universities";

export const mockUniversity: UniversityModel = {
  alpha_two_code: "PH",
  country: "Philippines",
  name: "University Test 1",
  domains: ["university.com"],
  web_pages: ["http://university.com"],
  "state-province": null,
};

export const mockUniversity2: UniversityModel = {
  alpha_two_code: "PH",
  country: "Philippines",
  name: "University Test 2",
  domains: ["university2.com"],
  web_pages: ["http://university2.com"],
  "state-province": null,
};
