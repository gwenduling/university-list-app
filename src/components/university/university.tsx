import { UniversityModel } from "../../models/universities";

import "./university.scss";

function University(props: Props) {
  return (
    <div className="university-component">
      <p className="name">{props.university.name}</p>
      <a
        className="link"
        href={props.university.web_pages[0]}
        target="_blank"
        rel="noreferrer"
        title={props.university.name}
      >
        {props.university.domains[0]}
      </a>
    </div>
  );
}

interface Props {
  university: UniversityModel;
}
export default University;
