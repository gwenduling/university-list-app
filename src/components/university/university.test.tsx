import University from "./university";

import { shallow } from "enzyme";
import { mockUniversity } from "../../test-data/university";

describe("Univesity Component", () => {
  it("should match the snapshot", () => {
    const wrapper = shallow(<University university={mockUniversity} />);
    expect(wrapper).toMatchSnapshot();
  });
});
