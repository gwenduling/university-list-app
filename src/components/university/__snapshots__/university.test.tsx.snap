// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Univesity Component should match the snapshot 1`] = `
<div
  className="university-component"
>
  <p
    className="name"
  >
    University Test 1
  </p>
  <a
    className="link"
    href="http://university.com"
    rel="noreferrer"
    target="_blank"
    title="University Test 1"
  >
    university.com
  </a>
</div>
`;
